### DISCUSSION TITLE
Is it possible to Add more drives/disks to existing minio stack?

### DISCUSSION THREAD

**Discussion by patachi** [minio/minio#19280] [Q&A] [answered]
Let's say we already have a Single-Node Multi-Drive with 2 drives, we ran out of space and we want to attach 2 more drives and expand the current 2x 8TB disk deployment to 4x 8TB disk deployment.

1. Is it even possible? (considering _minio_ is a **software** storage solution it shouldn't be hard-bounded to hardware)
2. Does it require explicit re-syncing of existing data to get back online or will it manage it in the background while serving existing data?
3. How to accomplish this? Any tips or requirements that should be considered?

**Comment by harshavardhana** [comment]
https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html#expand-minio-distributed

**Reply by patachi** [reply, marked as answer]
So there is no way to add more disks in Single-Node Multi-Drive deployment.

The only way to expand storage is possible in  Multi-Node Multi-Drive deployment.

And even in Multi-Node Multi-Drive mode, it is not possible to add more disks to the current server pool. 

The only way is to spin up a new server pool and add it to the cluster may be to decommission old pools, and you will left with a pile of unused old pool disks at your desk.

**Comment by drolex2** [comment]
I would like to challenge selected answer. The linked page talks about how to expand a distributed MinIO deployment. Patachi does not have a distributed deployment, and that page does not say that the storage in a single node cannot be expanded. Some details about the storage configuration are also missing. It's my understanding that a minimum of 4 drives are required for erasure coding, which would mean that Patachi is not using erasure coding. What is the reason that MinIO could not be restarted with additional drives with or without erasure coding? If the drives are pooled using LVM and MinIO is given one volume comprising of multiple drives, cannot the volume be expanded with more drives?
